docs/proposals/Periodic-Retrieval-of-Certificates-and-Keys-from-AKV.md

```diff
@@ -0,0 +1,121 @@
+
+Below is an example of certificate configuration for resource `KeyManagementProvider`. The version for the certificate should not be specified.
+
+```yaml
+apiVersion: config.ratify.deislabs.io/v1beta1
+kind: KeyManagementProvider
+metadata:
+name: keymanagementprovider-akv
+spec:
+type: azurekeyvault
+parameters:
+    vaultURI: https://${AKV_NAME}.vault.azure.net/
+    certificates:
+    - name: ${CERT_NAME}
+    tenantID: ${TENANT_ID}
+    clientID: ${IDENTITY_CLIENT_ID}
+```
+
+Below is an example of key configuration for resource `KeyManagementProvider`. The version for the key should not be specified.
+
+```yaml
+apiVersion: config.ratify.deislabs.io/v1beta1
+kind: KeyManagementProvider
+metadata:
+name: keymanagementprovider-akv
+spec:
+type: azurekeyvault
+parameters:
+    vaultURI: https://${AKV_NAME}.vault.azure.net/
+    keys:
+    - name: ${KEY_NAME}
+    tenantID: ${TENANT_ID}
+    clientID: ${IDENTITY_CLIENT_ID}
+```
+
+Below is an example to override the default retrieval interval:
+
+```yaml
+apiVersion: config.ratify.deislabs.io/v1beta1
+kind: KeyManagementProvider
+metadata:
+name: keymanagementprovider-akv
+spec:
+type: azurekeyvault
+parameters:
+    vaultURI: https://${AKV_NAME}.vault.azure.net/
+    keys:
+    - name: ${KEY_NAME}
+    tenantID: ${TENANT_ID}
+    clientID: ${IDENTITY_CLIENT_ID}
+    retrievalInterval: 4h
+```
+
+### Periodically reconciling versioned certificates or keys
+
+Users can specify the version for the certificate or key. In this case, reconciliation only takes place for the specified version, rather than retrieving all available versions for the certificate or key.
+
+Below is an example of versioned key for verifying Cosign signatures
+
+```yaml
+apiVersion: config.ratify.deislabs.io/v1beta1
+kind: KeyManagementProvider
+metadata:
+name: keymanagementprovider-akv
+spec:
+type: azurekeyvault
+parameters:
+    vaultURI: https://${AKV_NAME}.vault.azure.net/
+    keys:
+    - name: ${KEY_NAME}
+        version: ${KEY_VER}
+    tenantID: ${TENANT_ID}
+    clientID: ${IDENTITY_CLIENT_ID}
+```
+
+### Force an immediate retrieval of certificates and keys
+
+Users can update `KeyManagementProvider` resource and then use `kubectl apply` to apply it to trigger an immediate retrieval of certificates and keys.
```

Comment:
what does the user update? If they don't want to change the config, what is our guidance on a field to update that will trigger reconciliation?

Reply:
> what does the user update? If they don't want to change the config, what is our guidance on a field to update that will trigger reconciliation?

This is a good question. I would like to discuss this with you to see whether there is anything users can do. Or they just restart Ratify instance one by one, given 3 instances as the default for production

Reply:
updating a trivial field in metadata could trigger a reconcile, which can be a workaround.

Reply:
I would like to keep the metadata field update as an open question. Maybe users can just update the name of the resource?